web/packages/teleport/src/services/userEvent/types.ts

```diff
@@ -76,8 +81,64 @@ export enum IntegrationEnrollKind {
   MachineIDKubernetes = 'INTEGRATION_ENROLL_KIND_MACHINE_ID_KUBERNETES',
   EntraId = 'INTEGRATION_ENROLL_KIND_ENTRA_ID',
   DatadogIncidentManagement = 'INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT',
+  AwsIdentityCenter = 'INTEGRATION_ENROLL_KIND_AWS_IDENTITY_CENTER',
+}
+
+/**
+ * IntegrationEnrollStep defines configurable steps for an integration type.
+ */
+export enum IntegrationEnrollStep {
+  /**
+   * AWSIC steps defined for AWS Idenity Center plugin.
+   * Value matches with proto enums defined in the backend.
+   */
+  ConnectOidc = 'INTEGRATION_ENROLL_STEP_AWSIC_CONNECT_OIDC',
+  ImportResourceSetDefaultOwner = 'INTEGRATION_ENROLL_STEP_AWSIC_SET_ACCESSLIST_DEFAULT_OWNER',
+  IdentitySourceUploadSamlMetadata = 'INTEGRATION_ENROLL_STEP_AWSIC_UPLOAD_AWS_SAML_SP_METADATA',
+  ScimTestConnection = 'INTEGRATION_ENROLL_STEP_AWSIC_TEST_SCIM_CONNECTION',
+}
+
+/**
+ * IntegrationEnrollStatusCode defines status codes for a given
+ * integration configuration step event.
+ * Value matches with proto enums defined in the backend.
+ */
+export enum IntegrationEnrollStatusCode {
+  Success = 'INTEGRATION_ENROLL_STATUS_CODE_SUCCESS',
+  Skipped = 'INTEGRATION_ENROLL_STATUS_CODE_SKIPPED',
+  Error = 'INTEGRATION_ENROLL_STATUS_CODE_ERROR',
+  Aborted = 'INTEGRATION_ENROLL_STATUS_CODE_ABORTED',
 }
 
+/**
+ * IntegrationEnrollStepStatus defines fields for reporting
+ * integration configuration step event.
+ */
+export type IntegrationEnrollStepStatus = {
+  code: IntegrationEnrollStatusCode;
+  error?: string;
+};
```

Comment:
What would you say for defining this type as a discriminated union?
```suggestion
export type IntegrationEnrollStepStatus =
  | {
      code: Exclude<
        IntegrationEnrollStatusCode,
        IntegrationEnrollStatusCode.Error
      >;
    }
  | {
      code: IntegrationEnrollStatusCode.Error;
      error: string;
    };
```
This has some benefits over the optional field:
1. The error is now required for the error status code.
2. It prevents passing the error for non-error status codes.

Then I'd change the `emitEvent` signature to:
```ts
export function emitEvent(
  eventId: string,
  step: IntegrationEnrollStep,
  status: IntegrationEnrollStepStatus
)
```

Reply:
I like it + makes the input param types stronger. Many thanks for the suggestion https://github.com/gravitational/teleport/pull/51097/commits/ade358d8fb783ba820d08e73f30288ec6de7d12c